### PR TITLE
(feat): Add functionality to validate version and upated the provided version

### PR DESCRIPTION
**Pull request by imrajdas**
- [x] Added semantic version for regex
- [x] Added script to create the new versions from basetemplate
Signed-off-by: rajdas <mail.rajdas@gmail.com>

### CODE REVIEW

File: scripts/push.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+
+setup_git() {
+  git config --global user.email "travis@travis-ci.org"
+  git config --global user.name "Travis CI"
+  git remote set-url origin  https://${GITHUB_TOKEN}@github.com/rajdas98/community-charts.git > /dev/null 2>&1
```
Comment (chandankumar4) [inline scripts/push.sh]: Why we are using the **rajdas98** here?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+	# dependency to run gem pkg manger
+	sudo apt-get install rubygems
+	bash ./scripts/version_maker.sh
+
+.PHONY: push
+push:
+	@echo "---------git push to master-------"
+	bash ./scripts/push.sh
+
+
```
Comment (chandankumar4) [inline Makefile]: Delete the extra space/line

File: scripts/push.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+commit_website_files() {
+  git checkout master
+  git status
+  git add .
+  git commit --message "Travis build: $TRAVIS_BUILD_NUMBER"
```
Comment (chandankumar4) [inline scripts/push.sh]: Change the commit message to something like:- 
`Travis updates #<Build_id>: Version upgraded for chaos-charts`

File: scripts/push.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+
+setup_git() {
+  git config --global user.email "travis@travis-ci.org"
+  git config --global user.name "Travis CI"
+  git remote set-url origin  https://${GITHUB_TOKEN}@github.com/rajdas98/community-charts.git > /dev/null 2>&1
+}
+
+commit_website_files() {
```
Comment (chandankumar4) [inline scripts/push.sh]: Change the function name to: commit_website_files() -> `commit_updated_changes()`

File: scripts/version_maker.sh [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    # get the latest version from the template file
+    
+    eval $(yaml $file)
+    kind=$kind
+
+    if [ $kind == "ChaosExperiment" ]; then
+        newversion=$metadata_version
+    elif [ $kind == "ChartServiceVersion" ]; then
+        newversion=$spec_version 
+    fi
```
Comment (chandankumar4) [inline scripts/version_maker.sh]: Add comment for both cases

File: scripts/version_maker.sh [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+# add all the files to an array called files
+files=$(echo $FIND_CMD | tr " " "\n")
+
+# looping from all the files
+for file in $files
+do
+    # get the latest version from the template file
+    
+    eval $(yaml $file)
```
Comment (ksatchit) [inline scripts/version_maker.sh:31]: any particular reason we are using eval?

Comment (ksatchit) [reply scripts/version_maker.sh:31]: do you think we need $? checks ?

**Review by ksatchit — APPROVED** [review]
lgtm

**Pull request merged**